File: Fruit Into Baskets.cpp
class Solution {
public:
    int totalFruit(vector<int>& fruits) {
       int lastfruit = -1, secondlast = -1;
        int lastcount=0, currmax=0, maxf=0;
        for(int fruit : fruits)
       {
          if(fruit==lastfruit || fruit==secondlast)
              currmax+=1;
            else
                currmax = lastcount+1;
            if(fruit==lastfruit)
                lastcount+=1;
            else
                lastcount = 1;
            if(fruit!=lastfruit){
                secondlast = lastfruit;
                lastfruit  = fruit;
            }
            maxf = max(currmax,maxf);
      }
        return maxf;
    }
};

File: Fruit Into Baskets.py
class Solution:
    def totalFruit(self, fruits: List[int]) -> int:
        left = 0
        maxfruit = 0
        unique = defaultdict(int)
        count = 0

        for right in range(len(fruits)):
            if fruits[right] not in unique or unique[fruits[right]]==0:
                count += 1
            unique[fruits[right]] += 1
            while count > 2:
                unique[fruits[left]] -= 1
                if unique[fruits[left]] == 0:
                    count -= 1
                left += 1
            maxfruit = max(maxfruit, right - left + 1)
        
        return maxfruit
                 
